staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -31,147 +31,161 @@ import (
 )
 
 // NestedFieldCopy returns a deep copy of the value of a nested field.
-// false is returned if the value is missing.
-// nil, true is returned for a nil field.
-func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{}, bool) {
-	val, ok := nestedFieldNoCopy(obj, fields...)
-	if !ok {
-		return nil, false
+// An error is returned if the value is missing.
+// No error  is returned for a nil field.
+func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{}, error) {
+	val, err := nestedFieldNoCopy(obj, fields...)
+	if err != nil {
+		return nil, err
 	}
-	return runtime.DeepCopyJSONValue(val), true
+	return runtime.DeepCopyJSONValue(val), nil
 }
 
-func nestedFieldNoCopy(obj map[string]interface{}, fields ...string) (interface{}, bool) {
+func nestedFieldNoCopy(obj map[string]interface{}, fields ...string) (interface{}, error) {
 	var val interface{} = obj
 	for _, field := range fields {
 		if m, ok := val.(map[string]interface{}); ok {
 			val, ok = m[field]
 			if !ok {
-				return nil, false
+				return nil, fmt.Errorf("field does not exist")
 			}
 		} else {
-			// Expected map[string]interface{}, got something else
-			return nil, false
+			return nil, fmt.Errorf("%v is not of the type map[string]interface{}", val)
```

Comment:
Here and in other places: it would be extremely helpful to actually get the type of `val` in the error message too. Something like `fmt.Errorf("%v is of the type %T, expected map[string]interface{}", val, val)`.

Also see my PR which I'll probably close: https://github.com/kubernetes/kubernetes/pull/55297

Reply:
Done.